File: demo_laya_js/src/ui/LetoBannerDemo.js
export default class LetoBannerDemo extends Laya.Scene {
    constructor() {
        super();
        
        //加载场景文件
        this.loadScene("test/LetoBannerDemo.scene");

        // event
        this.btnBack.on(Laya.Event.CLICK, this, this.onBackClicked)
        this.btnLoad.on(Laya.Event.CLICK, this, this.onLoadClicked)
        this.btnReady.on(Laya.Event.CLICK, this, this.onReadyClicked)
        this.btnShow.on(Laya.Event.CLICK, this, this.onShowClicked)
        this.btnRemove.on(Laya.Event.CLICK, this, this.onRemoveClicked)
        this.btnReshow.on(Laya.Event.CLICK, this, this.onReshowClicked)
        this.btnHide.on(Laya.Event.CLICK, this, this.onHideClicked)
    }

    adId() {
        return 1
    }

    onBackClicked() {
        Laya.Scene.open('test/LetoMainDemo.scene')
    }

    onLoadClicked() {
        LTBannerJSSDK.load(this.adId());
    }

    onReadyClicked() {
        var hasReady = LTBannerJSSDK.hasAdReady(this.adId());
        LTJSSDK.printLog("LetoAdBannerDemo::checkReady() " + (hasReady ? "Ready" : "NO"));
    }

    onShowClicked() {
        LTBannerJSSDK.show(this.adId());
    }

    onRemoveClicked() {
        LTBannerJSSDK.remove(this.adId());
    }

    onReshowClicked() {
        LTBannerJSSDK.reShow(this.adId());
    }

    onHideClicked() {
        LTBannerJSSDK.hide(this.adId());
    }

    onBannerAdLoaded(adId, adInfo) {
        LTJSSDK.printLog(`LetoAdBannerDemo::onBannerAdLoaded(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onBannerAdLoadFail(adId, errMsg) {
        LTJSSDK.printLog("LetoAdBannerDemo::onBannerAdLoadFail(" + adId + ", " + errMsg + ")");   
    }

    onBannerAdClick(adId, adInfo) {
        LTJSSDK.printLog(`LetoAdBannerDemo::onBannerAdClick(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onBannerAdShow(adId, adInfo) {
        LTJSSDK.printLog(`LetoAdBannerDemo::onBannerAdShow(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onBannerAdHide(adId, adInfo) {
        LTJSSDK.printLog(`LetoAdBannerDemo::onBannerAdHide(${adId}, ${JSON.stringify(adInfo)})`);
    }

    onBannerAdClose(adId, adInfo) {
        LTJSSDK.printLog(`LetoAdBannerDemo::onBannerAdClose(${adId}, ${JSON.stringify(adInfo)})`);
    }
}

File: demo_unity/Assets/Scripts/LetoBannerDemo.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.SceneManagement;
using LetoAd;

public class LetoBannerDemo : MonoBehaviour
{
    private ILTBannerCSSDK _letoSdk;
    private int _adId = 1;

    void Awake() {
        _letoSdk = new LTBannerCSSDK();
        _letoSdk.setAdListener(new EventListener());
    }

    public void OnBackClicked() {
        SceneManager.LoadScene("LetoMainDemo");
    }

    public void OnLoadClicked() {
        _letoSdk.load(_adId);
    }

    public void OnReadyClicked() {
        bool ready = _letoSdk.isAdReady(_adId);
        Debug.Log("OnReadyClicked, ready: " + ready);
    }

    public void OnShowClicked() {
        _letoSdk.show(_adId);
    }

    public void OnRemoveClicked() {
        _letoSdk.remove(_adId);
    }

    public void OnReshowClicked() {
        _letoSdk.reShow(_adId);
    }

    public void OnHideClicked() {
        _letoSdk.hide(_adId);
    }

    class EventListener : ILTBannerListener {
        public void onBannerLoaded(int adId, LTAdInfo adInfo) {
            Debug.Log("onBannerLoaded: " + adId);
        }

        public void onBannerLoadFail(int adId, string errMsg) {
            Debug.Log("onBannerLoadFail: " + adId + ", errMsg: " + errMsg);
        }

        public void onBannerClick(int adId, LTAdInfo adInfo) {
            Debug.Log("onBannerClick: " + adId);
        }

        public void onBannerShow(int adId, LTAdInfo adInfo) {
            Debug.Log("onBannerShow: " + adId);
        }

        public void onBannerHide(int adId, LTAdInfo adInfo) {
            Debug.Log("onBannerHide: " + adId);
        }

        public void onBannerClose(int adId, LTAdInfo adInfo) {
            Debug.Log("onBannerClose: " + adId);
        }
    }
}
